File: skills/astra-toolkit/clients/typescript/examples/tables-create-type-exists.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get a database
const client = new DataAPIClient();
const database = client.db(process.env.ASTRA_DB_API_ENDPOINT!, {
  token: process.env.ASTRA_DB_APPLICATION_TOKEN!,
});

// Drop a user-defined type
(async function () {
  await database.createType("member", {
    definition: {
      fields: {
        name: "text",
        is_active: "boolean",
        date_joined: "date",
      },
    },
    ifNotExists: true,
  });
})();

File: skills/astra-toolkit/clients/python/examples/tables-create-type-exists.py
import os
from astrapy import DataAPIClient
from astrapy.info import (
    ColumnType,
    CreateTypeDefinition,
    TableScalarColumnTypeDescriptor,
)

# Get a database
client = DataAPIClient()
database = client.get_database(
    os.environ["ASTRA_DB_API_ENDPOINT"], token=os.environ["ASTRA_DB_APPLICATION_TOKEN"]
)

# Create a user-defined type
type_definition = CreateTypeDefinition(
    fields={
        "name": TableScalarColumnTypeDescriptor(ColumnType.TEXT),
        "is_active": TableScalarColumnTypeDescriptor(ColumnType.BOOLEAN),
        "date_joined": TableScalarColumnTypeDescriptor(ColumnType.DATE),
    },
)
database.create_type(
    "member", definition=type_definition, if_not_exists=True
)
